File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;

namespace WindowsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        Dictionary<string, string> productEncodeTable;
        Dictionary<string, string> ProductEncodeTable {
            get {
                if(productEncodeTable == null)
                    productEncodeTable = new Dictionary<string, string>();
                return productEncodeTable;
            }
        }
        Dictionary<string, string> categoryEncodeTable;
        Dictionary<string, string> CategoryEncodeTable {
            get {
                if(categoryEncodeTable == null)
                    categoryEncodeTable = new Dictionary<string, string>();
                return categoryEncodeTable;
            }
        }

        void CreateEncodeTables() {
            int productCounter = 1;
            foreach(object value in fieldProductName.GetUniqueValues()) {
                ProductEncodeTable.Add(value.ToString(), "Products: " + productCounter++);
            }

            int categoryCounter = 1;
            foreach(object value in fieldCategoryName.GetUniqueValues()) {
                CategoryEncodeTable.Add(value.ToString(), "Category: " + categoryCounter++);
            }
        }

        private void Form1_Load(object sender, EventArgs e) {
            this.productReportsTableAdapter.Fill(this.nwindDataSet.ProductReports);
            pivotGridControl1.OptionsChartDataSource.ProvideRowFieldValuesAsType = typeof(string);
            pivotGridControl1.OptionsChartDataSource.ProvideColumnFieldValuesAsType = typeof(string);
            pivotGridControl1.OptionsChartDataSource.ProvideCellValuesAsType = typeof(int);
            CreateEncodeTables();
            chartControl1.DataSource = pivotGridControl1;
            fieldCategoryName.CollapseAll();
            fieldCategoryName.ExpandValue("Produce");
        }

        #region #CustomChartDataSourceData
        private void pivotGridControl1_CustomChartDataSourceData(object sender, 
        PivotCustomChartDataSourceDataEventArgs e) {
            if(e.ItemType == PivotChartItemType.RowItem) {
                if(e.FieldValueInfo.Field == fieldCategoryName) {
                    e.Value = CategoryEncodeTable[e.FieldValueInfo.Value.ToString()];
                } else if(e.FieldValueInfo.Field == fieldProductName) {
                    string product =  
                        ProductEncodeTable[e.FieldValueInfo.Value.ToString()];
                    string category = 
                    CategoryEncodeTable[e.FieldValueInfo.GetHigherLevelFieldValue(fieldCategoryName).ToString()];
                    e.Value = product + '[' + category + ']';
                }
            }
            if(e.ItemType == PivotChartItemType.ColumnItem) {
                if(e.FieldValueInfo.ValueType == PivotGridValueType.GrandTotal) {
                    e.Value = "Total Sales";
                }
            }
            if(e.ItemType == PivotChartItemType.CellItem) {
                e.Value = Math.Round(Convert.ToDecimal(e.CellInfo.Value), 0);
            }
        }
        #endregion #CustomChartDataSourceData
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

'INSTANT VB NOTE: The field productEncodeTable was renamed since Visual Basic does not allow fields to have the same name as other class members:
		Private productEncodeTable_Renamed As Dictionary(Of String, String)
		Private ReadOnly Property ProductEncodeTable() As Dictionary(Of String, String)
			Get
				If productEncodeTable_Renamed Is Nothing Then
					productEncodeTable_Renamed = New Dictionary(Of String, String)()
				End If
				Return productEncodeTable_Renamed
			End Get
		End Property
'INSTANT VB NOTE: The field categoryEncodeTable was renamed since Visual Basic does not allow fields to have the same name as other class members:
		Private categoryEncodeTable_Renamed As Dictionary(Of String, String)
		Private ReadOnly Property CategoryEncodeTable() As Dictionary(Of String, String)
			Get
				If categoryEncodeTable_Renamed Is Nothing Then
					categoryEncodeTable_Renamed = New Dictionary(Of String, String)()
				End If
				Return categoryEncodeTable_Renamed
			End Get
		End Property

		Private Sub CreateEncodeTables()
			Dim productCounter As Integer = 1
			For Each value As Object In fieldProductName.GetUniqueValues()
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: ProductEncodeTable.Add(value.ToString(), "Products: " + productCounter++);
				ProductEncodeTable.Add(value.ToString(), "Products: " & productCounter)
				productCounter += 1
			Next value

			Dim categoryCounter As Integer = 1
			For Each value As Object In fieldCategoryName.GetUniqueValues()
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: CategoryEncodeTable.Add(value.ToString(), "Category: " + categoryCounter++);
				CategoryEncodeTable.Add(value.ToString(), "Category: " & categoryCounter)
				categoryCounter += 1
			Next value
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			Me.productReportsTableAdapter.Fill(Me.nwindDataSet.ProductReports)
			pivotGridControl1.OptionsChartDataSource.ProvideRowFieldValuesAsType = GetType(String)
			pivotGridControl1.OptionsChartDataSource.ProvideColumnFieldValuesAsType = GetType(String)
			pivotGridControl1.OptionsChartDataSource.ProvideCellValuesAsType = GetType(Integer)
			CreateEncodeTables()
			chartControl1.DataSource = pivotGridControl1
			fieldCategoryName.CollapseAll()
			fieldCategoryName.ExpandValue("Produce")
		End Sub

		#Region "#CustomChartDataSourceData"
		Private Sub pivotGridControl1_CustomChartDataSourceData(ByVal sender As Object, ByVal e As PivotCustomChartDataSourceDataEventArgs) Handles pivotGridControl1.CustomChartDataSourceData
			If e.ItemType = PivotChartItemType.RowItem Then
				If e.FieldValueInfo.Field Is fieldCategoryName Then
					e.Value = CategoryEncodeTable(e.FieldValueInfo.Value.ToString())
				ElseIf e.FieldValueInfo.Field Is fieldProductName Then
					Dim product As String = ProductEncodeTable(e.FieldValueInfo.Value.ToString())
					Dim category As String = CategoryEncodeTable(e.FieldValueInfo.GetHigherLevelFieldValue(fieldCategoryName).ToString())
					e.Value = product & "["c & category & "]"c
				End If
			End If
			If e.ItemType = PivotChartItemType.ColumnItem Then
				If e.FieldValueInfo.ValueType = PivotGridValueType.GrandTotal Then
					e.Value = "Total Sales"
				End If
			End If
			If e.ItemType = PivotChartItemType.CellItem Then
				e.Value = Math.Round(Convert.ToDecimal(e.CellInfo.Value), 0)
			End If
		End Sub
		#End Region ' #CustomChartDataSourceData
	End Class
End Namespace